apis/apiextensions/v1alpha1/composition_types.go

```diff
@@ -48,19 +48,21 @@ var (
 
 // CompositionSpec specifies the desired state of the definition.
 type CompositionSpec struct {
-	// From refers to the type that this composition is compatible. The values
-	// for the underlying resources will be fetched from the instances of the
-	// From.
+	// CompositeTypeRef refers to the type that this composition is compatible.
+	// The values for the underlying resources will be fetched from the
+	// instances of the referenced composite type.
 	// +immutable
-	From TypeReference `json:"from"`
+	CompositeTypeRef TypeReference `json:"compositeTypeRef"`
 
-	// To is the list of target resources that make up the composition.
-	To []ComposedTemplate `json:"to"`
+	// Resources is the list of resource templates that will be used when a
+	// composite resource referring to this composition is created.
+	Resources []ComposedTemplate `json:"resources"`
 
 	// WriteConnectionSecretsToNamespace specifies the namespace in which the
 	// connection secrets of composite resource dynamically provisioned using
 	// this composition will be created.
-	WriteConnectionSecretsToNamespace string `json:"writeConnectionSecretsToNamespace"`
+	// +optional
+	WriteConnectionSecretsToNamespace string `json:"writeConnectionSecretsToNamespace,omitempty"`
```

Comment:
I can't comment on the existing thread about this for some reason, but I tend to agree with @muvaf that we should make this a `*string` since it's now optional. I believe it's only used in one place, so updating the corresponding controller code should hopefully be straightforward.

Reply:
Agreed, responded in thread above.